rexclient/compression.go

```diff
@@ -0,0 +1,39 @@
+package rexclient
+
+import (
+	"bytes"
+	"net/http"
+	"strings"
+
+	"github.com/bazelbuild/remote-apis-sdks/go/pkg/uploadinfo"
+)
+
+// DetectArchiveUploads is an UploadCompressionPredicate that attempts to detect whether
+// uploads are already archives, either based on their initial few bytes or known file extensions.
+// It is a best-effort classifier that may not recognise every possible type.
+func DetectArchiveUploads(ue *uploadinfo.Entry) bool {
```

Comment:
They (probably reasonably) didn't want this upstream in the SDK, but the API lets us determine compression from this repo now.